pkg/instance/instance.go

```diff
@@ -59,3 +70,29 @@ func (i *Info) UpgradeRequired() bool {
 	// fully deconfigured before being configured by the current version.
 	return true
 }
+
+// getIPv4Address returns an IPv4 address associated with the given address. An error will be thrown if given
+// an IPv6 address or a DNS address that does not resolve to an IPv4 network address.
+func getIPv4Address(address string) (string, error) {
+	if ip := net.ParseIP(address); ip != nil {
+		// Address is either an IPv6 or IPv4 address
+		ipv4 := ip.To4()
+		if ipv4 == nil {
+			return "", errors.Errorf("IPv6 network address not supported: %s", ip.String())
```

Comment:
Nit: I would rather say not an IPv4 address given the [doc](https://pkg.go.dev/net#IP.To4) says:
>If ip is not an IPv4 address, To4 returns nil.

Reply:
I am not certain what you mean here. I assume you mean not specifying `IPv6` in the error message. After the `parseIP` call we know that `To4` is called on either an IPv4 or an IPv6 address (whether mapped or unmapped), so it is accurate to do so IMO.

Reply:
Not an IPv4 address needn't imply a IPv6 address is the point I am making. So it is safer to say `not an IPv4 address`.